pyannote/audio/utils/powerset.py

```diff
@@ -24,8 +24,9 @@
 # Hervé BREDIN - https://herve.niderb.fr
 # Alexis PLAQUET
 
-from functools import cached_property
```

Comment:
`lru_cache` is imported but never used.
Two things: 
* I recommended you install the precommit hooks described [here](https://github.com/pyannote/pyannote-audio#development) to avoid this kind of thing and keep consistent with pyannote code style
* Use it :) but first make sure that `perm_ml` is actually hashable. For `lru_cache` to be usable you might actually need to make it a `tuple` of `int`s.

Reply:
I am wondering whether it would not make more sense to precompute all those permutations and make them an attribute of `Powerset` or a `register` like we already did for `cardinality`.

Reply:
Commit 46819fe implements that. The conversion seems to have way better performance now ! (shocking :) )